src/reanimated2/js-reanimated/JSReanimated.ts

```diff
@@ -6,12 +6,16 @@ import {
   Value3D,
   ValueRotation,
 } from '../commonTypes';
+import { mockedRequestAnimationFrame } from '../utils';
 import { WebSensor } from './WebSensor';
 
 // In Node.js environments (like when static rendering with Expo Router)
 // requestAnimationFrame is unavailable, so we use setImmediate.
-const scheduleOnUI =
-  globalThis.requestAnimationFrame || globalThis.setImmediate;
+const scheduleOnUI = globalThis.requestAnimationFrame
+  ? isJest()
+    ? mockedRequestAnimationFrame
+    : globalThis.requestAnimationFrame
+  : setImmediate;
```

Comment:
1. First, can we handle Jest first and then the rest?
```ts
const scheduleOnUI = isJest()
  ? mockedRequestAnimationFrame
  : globalThis.requestAnimationFrame || globalThis.setImmediate;
```
2. Can we add a new platform checker `isStaticWeb()`? This way we could write it this way:
```ts
const scheduleOnUI = isJest()
  ? mockedRequestAnimationFrame
  : isStaticWeb()
  ? globalThis.setImmediate
  : globalThis.requestAnimationFrame;
```

Reply:
1. As far as I understand in this environment `useAnimatedFrame` is unavailable in Jest also and we want to use `setImmediate` there. With this implementation we would always use the mock - maybe it's desired - @javascripter could you give more insight on that?

2. This is hard, we have to discuss it tomorrow.

Reply:
I used setImmediate in the static rendering environment simply because I tested it in my environment and it seemed to work well for me. As @satya164 said, scheduleOnUI probably doesn't need to execute in that environment so I don't have a concrete reason to prefer setImmediate over the mock in Jest. Always using the mock in Jest seems a reasonable approach to me.  @tjzel

Reply:
We actually had a quite fruitful discussion about this and we want to go a step further and improve how Reanimated behaves during Static Renders. This PR can stay as a workaround for the moment, but we will open a proper one in the near future.